experimental/dds/tree2/src/test/feature-libraries/default-field-kinds/optionalField.spec.ts

```diff
@@ -348,17 +374,108 @@ describe("optionalField", () => {
 				},
 			];
 
-			assertMarkListEqual(optionalFieldIntoDelta(change1.change, deltaFromChild1), expected);
+			const expectedRepairData = new Map<FieldKey, Delta.MarkList>([
+				[
+					brand("repair-data-0"),
+					[
+						{
+							type: Delta.MarkType.MoveIn,
+							count: 1,
+							moveId: brand(0),
+						},
+					],
+				],
+			]);
+
+			const { repairDataBuilder } = makeRepairDataBuilder();
+			assertMarkListEqual(
+				optionalFieldIntoDelta(
+					change1.change,
+					deltaFromChild1,
+					repairDataBuilder,
+					idAllocatorFromMaxId(),
+				),
+				expected,
+			);
+			assert.deepEqual(repairDataBuilder.marks, expectedRepairData);
+		});
+
+		it("can be converted to a delta when replacing content", () => {
+			const change: TaggedChange<OptionalChangeset> = tagChange(
+				{
+					fieldChange: {
+						id: brand(1),
+						newContent: { set: testTree("tree1") },
+						wasEmpty: false,
+					},
+				},
+				change1Revision,
+			);
+
+			const expected: Delta.MarkList = [
+				{ type: Delta.MarkType.MoveOut, count: 1, moveId: brand(0), isDelete: true },
+				{
+					type: Delta.MarkType.Insert,
+					content: [testTreeCursor("tree1")],
+				},
+			];
+
+			const expectedRepairData = new Map<FieldKey, Delta.MarkList>([
+				[
+					brand("repair-data-0"),
+					[
+						{
+							type: Delta.MarkType.MoveIn,
+							count: 1,
+							moveId: brand(0),
+							isDelete: true,
+						},
+					],
+				],
+			]);
+
+			const { repairDataBuilder } = makeRepairDataBuilder();
+			assertMarkListEqual(
+				optionalFieldIntoDelta(
+					change.change,
+					() => assert.fail("Should not need to convert child changes"),
+					repairDataBuilder,
+					idAllocatorFromMaxId(),
+				),
+				expected,
+			);
+			assert.deepEqual(repairDataBuilder.marks, expectedRepairData);
 		});
 
 		it("can be converted to a delta when restoring content", () => {
 			const expected: Delta.MarkList = [
-				{ type: Delta.MarkType.Delete, count: 1 },
+				{ type: Delta.MarkType.MoveOut, count: 1, moveId: brand(0), isDelete: true },
 				{ type: Delta.MarkType.Insert, content: [testTreeCursor("tree1")] },
 			];
 
-			const actual = optionalFieldIntoDelta(revertChange2, deltaFromChild1);
+			const expectedRepairData = new Map<FieldKey, Delta.MarkList>([
+				[
```

Comment:
There should also be a move-out from the repair data.

Reply:
The move out is part of `expected`. I check them separately because the move in gets added to the `repairDataMarks` map via the accumulator.

Reply:
No, I was referring to the MoveOut that would be necessary in `expectedRepairData` for the sake of restoring the content. This falls under the "do we support revive as part of this PR?" bucket.